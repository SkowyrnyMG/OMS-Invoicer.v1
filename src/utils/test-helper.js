import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
