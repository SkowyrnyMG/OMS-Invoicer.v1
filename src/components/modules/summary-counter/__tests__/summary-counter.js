import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import SummaryCounter from '../summary-counter';

describe('SummaryCounter', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<SummaryCounter title='test-title' counter={1} />);
  });
});
