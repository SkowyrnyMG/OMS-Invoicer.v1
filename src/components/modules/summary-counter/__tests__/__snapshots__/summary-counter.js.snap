// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCounter should display in the document with correct styles 1`] = `
.c1 {
  margin-bottom: 0.5rem;
  font-size: 20px;
  font-weight: 400;
  text-transform: uppercase;
  color: #446DF6;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center !important;
  -webkit-box-align: center !important;
  -ms-flex-align: center !important;
  align-items: center !important;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 12rem;
  text-align: center;
}

.c0:not(:first-of-type) {
  border-left: 1px solid #F0F0FD;
}

@media (max-width:850px) {
  .c0 {
    border-bottom: 1px solid #F0F0FD;
  }
}

<div
  class="c0"
>
  <h5
    class="c1"
  >
    test-title
  </h5>
  <p>
    1
  </p>
</div>
`;
